app/components/selection/handle.element.js

```diff
@@ -0,0 +1,184 @@
+import $ from 'blingblingjs'
+import { HandleStyles } from '../styles.store'
+
+export class Handle extends HTMLElement {
+
+	constructor() {
+		super()
+    this.$shadow = this.attachShadow({mode: 'closed'})
+    this.styles = [HandleStyles]
+	}
+
+  connectedCallback() {
+    this.$shadow.adoptedStyleSheets = this.styles
+		this.$shadow.innerHTML = this.render();
+		
+		this.button = this.$shadow.querySelector('button')
+		this.button.addEventListener('pointerdown', this.on_element_resize_start.bind(this))
+
+		this.placement = this.getAttribute('placement')
+	}
+
+	static get observedAttributes() {
+    return ['placement']
+  }
+
+	attributeChangedCallback(name, oldValue, newValue) {
+    if (name === 'placement') {
+			this.placement = newValue
+		}
+  }
+
+	on_element_resize_start(e) {
+		e.preventDefault()
+		e.stopPropagation()
+
+		if (e.button !== 0) return
+
+		const placement = this.placement
+		const handlesEl = e.path.find(el => el.tagName === 'VISBUG-HANDLES')
+		const nodeLabelId = handlesEl.getAttribute('data-label-id')
+		const [sourceEl] = $(`[data-label-id="${nodeLabelId}"]`)
+
+		if (!sourceEl) return
+
+		const { x: initialX, y: initialY } = e
+		const initialStyle = getComputedStyle(sourceEl)
+		const initialWidth = parseFloat(initialStyle.width)
+		const initialHeight = parseFloat(initialStyle.height)
+		const initialTransform = new DOMMatrix(initialStyle.transform)
+
+		const originalElTransition = sourceEl.style.transition
+		const originalDocumentCursor = document.body.style.cursor
+		const originalDocumentUserSelect = document.body.style.userSelect
+		sourceEl.style.transition = 'none'
+		document.body.style.cursor = getComputedStyle(this).getPropertyValue('--cursor')
+		document.body.style.userSelect = 'none'
+
+		document.addEventListener('pointermove', on_element_resize_move)
+
+		function on_element_resize_move(e) {
+			e.preventDefault()
+			e.stopPropagation()
+
+			const newX = clamp(0, e.clientX, document.documentElement.clientWidth)
+			const newY = clamp(0, e.clientY, document.documentElement.clientHeight)
+		
+			const diffX = newX - initialX
+			const diffY = newY - initialY
+
+			switch (placement) {
+				case 'top-start': {
+					const newWidth = initialWidth - diffX
+					const newHeight = initialHeight - diffY
+					const newTranslate = initialTransform.translate(diffX, diffY).transformPoint()
+
+					requestAnimationFrame(() => {
+						sourceEl.style.width = `${newWidth}px`
+						sourceEl.style.height = `${newHeight}px`
+						sourceEl.style.transform = `translate(${newTranslate.x}px, ${newTranslate.y}px)`
```

Comment:
very clever to transform here! awesome UX attention to detail here, really sells the illusion. the same technique would probably support margin and padding visuals being grab-able 🤔 
 
also cool you used the DOMMatrix to do the math 🤘🏻 

not two things i would have reached for immediately, but find they totally rock for this scenario. cool math and programming plus a great user experience. 😍

Reply:
Jake deserves the credit for `DOMMatrix` 😄 https://youtu.be/VdNzD4lhidw

Reply:
i was wonderin!!! seemed like appropriate timing